pkg/blockatlas/client.go

```diff
@@ -15,10 +15,12 @@ import (
 )
 
 type Request struct {
-	BaseUrl      string
-	Headers      map[string]string
-	HttpClient   *http.Client
-	ErrorHandler func(res *http.Response, uri string) error
+	BaseUrl            string
+	Headers            map[string]string
+	HttpClient         *http.Client
+	ErrorHandler       func(res *http.Response, uri string) error
+	CacheHandler       func(method string, url string, body io.Reader, result interface{}) error
```

Comment:
is there a reason to have `CacheHandler`? I would assume most requests would just need to specify expiration time and not need for custom handlers or when handler might different?